### PR TITLE
19 configurable scoring criteria

### PR DESCRIPTION
**Pull request by kchetans**
scoring output customized by category or by feature, through config file or cmd line

eg. 
1. Through cmd line
./sbomqs score --filepath ./samples/sbomqs.cdx-modtool.json --feature doc-licence
SBOM Quality Score:0.0  components:14   ./samples/sbomqs.cdx-modtool.json
+----------+----------------------+----------+--------------------------------+
| CATEGORY |       FEATURE        |  SCORE   |              DESC              |
+----------+----------------------+----------+--------------------------------+
| Sharing  | Doc sharable license | 0.0/10.0 | doc has a sharable license     |
|          |                      |          | free 0 :: of 0                 |
+----------+----------------------+----------+--------------------------------+

2. Through config file
   a. Generate sbomqs config.yaml file
       eg../sbomqs generate
  b.  use the configuration file to enable through category and feature
       eg. ./sbomqs score --filepath ./samples/sbomqs.cdx-modtool.json --configpath config.yaml

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+category:
+    - name: NTIA-minimum-elements
+      enabled: true
+      weight: 100
```
Comment (surendrapathak) [inline config.yaml:4]: Please remove weights for this version because it has no impact

**Review by surendrapathak — CHANGES_REQUESTED** [review]

File: features.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
+category:
```
Comment (riteshnoronha) [inline features.yaml:1]: @kchetans  please remove this file.

File: pkg/scorer/scorer.go [hunk 3]
```diff
@@ -66,13 +67,25 @@ func (s *Scorer) Score() Scores {
 	}
 	scores := newScores()
 
-	for _, cr := range criteria {
-		score := cr(s.doc)
-		if s.category != "" && s.category == score.Category() {
-			scores.addScore(score)
-		} else if s.category == "" {
-			scores.addScore(score)
+	for key, cr := range criteria {
```
Comment (riteshnoronha) [inline pkg/scorer/scorer.go:70]: This logic needs to be simplified, maybe split out the common functionality .

**Review by riteshnoronha — CHANGES_REQUESTED** [review]

**Review by riteshnoronha — APPROVED** [review]

File: cmd/generate.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
+/*
+Copyright © 2023 NAME HERE <EMAIL ADDRESS>
```
Comment (surendrapathak) [inline cmd/generate.go:2]: Fix Copyright for Interlynk

**Review by surendrapathak — APPROVED** [review]
Copyright will need fixes. The rest looks good to me.

**Pull request merged**